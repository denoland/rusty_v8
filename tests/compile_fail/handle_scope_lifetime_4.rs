// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.
use std::pin::pin;

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let scope1 = pin!(v8::HandleScope::new(&mut isolate));
  let mut scope1 = scope1.init();

  let context = v8::Context::new(&mut scope1, v8::ContextOptions::default());
  let mut context_scope = v8::ContextScope::new(&mut scope1, context);

  let mut _scope3 = {
    v8::scope!(scope, &mut context_scope);
    v8::EscapableHandleScope::new(scope)
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
