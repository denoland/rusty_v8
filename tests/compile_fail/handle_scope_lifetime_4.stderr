error[E0597]: `scope` does not live long enough
  --> tests/compile_fail/handle_scope_lifetime_4.rs:13:5
   |
12 |   let mut _scope3 = {
   |       ----------- borrow later stored here
13 |     v8::scope!(scope, &mut context_scope);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     binding `scope` declared here
14 |     v8::EscapableHandleScope::new(scope)
15 |   };
   |   - `scope` dropped here while still borrowed
   |
   = note: this error originates in the macro `v8::scope` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0597]: `scope` does not live long enough
  --> tests/compile_fail/handle_scope_lifetime_4.rs:13:5
   |
12 |   let mut _scope3 = {
   |       ----------- borrow later stored here
13 |     v8::scope!(scope, &mut context_scope);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     binding `scope` declared here
14 |     v8::EscapableHandleScope::new(scope)
15 |   };
   |   - `scope` dropped here while still borrowed
   |
   = note: this error originates in the macro `v8::scope` (in Nightly builds, run with -Z macro-backtrace for more info)
